### PR TITLE
make IpcApp, IpcHost, NativeApp, NativeHost @public

### PR DESCRIPTION
**Pull request by kabentley**

### CODE REVIEW

File: core/backend/src/NativeHost.ts [hunk 1]
```diff
@@ -179,7 +179,8 @@ class NativeAppHandler extends IpcHandler implements NativeAppFunctions {
   }
 }
 
-/** @beta */
+/** Options for [[NativeHost.startup]]
+ * @public */
 export interface NativeHostOpts extends IpcHostOpts {
   nativeHost?: {
     /** Application named. Used to name settings file */
```
Comment (pmconne) [inline core/backend/src/NativeHost.ts:186]: What does it do if applicationName is undefined?

File: core/frontend/src/IpcApp.ts [hunk 2]
```diff
@@ -30,15 +30,15 @@ export type PromiseReturnType<T extends AsyncFunction> = T extends (...args: any
 
 /**
  * Options for [[IpcApp.startup]]
- * @beta
+ * @public
```
Comment (pmconne) [inline core/frontend/src/IpcApp.ts:33]: `[[IpcApp.startup]]` links to an internal API.

**Review by pmconne — APPROVED** [review]
One minor doc issue.

File: core/frontend/src/IpcApp.ts (hunk 2 above)
Comment (kabentley) [reply core/frontend/src/IpcApp.ts:33]: I decided to make `IpcApp.startup` public with a note that it isn't intended to be called directly.

Comment (pmconne) [reply core/frontend/src/IpcApp.ts:33]: Do you intend to push that change? Current diff shows you added `@internal` to IpcApp.startup.

Comment (kabentley) [reply core/frontend/src/IpcApp.ts:33]: pushed

**Pull request merged**